enhancements/storage/vsphere-storage-check.md

```diff
@@ -0,0 +1,228 @@
+---
+title: vsphere-monitoring-operator
+authors:
+  - "@jsafrane"
+reviewers:
+  - "@hekumar"
+  - "@jspeed"
+  - "@jcpowermac"
+  - # someone from installer
+approvers:
+  - TBD
+creation-date: 2020-11-10
+last-updated: 2020-11-10
+status: provisional
+see-also:
+replaces:
+superseded-by:
+---
+
+# vSphere monitoring for OCP clusters
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+Below we propose to check if OpenShift clusters installed into vSphere environment is / can be fully functional by introducing a new operator and periodical checks.
+
+## Motivation
+
+Our support frequently faces OCP clusters on vSphere that is either mis-configured.
+
+* OCP lacks necessary permissions: [#1869035](https://bugzilla.redhat.com/show_bug.cgi?id=1869035) [#1879959](https://bugzilla.redhat.com/show_bug.cgi?id=1879959). [#1852304](https://bugzilla.redhat.com/show_bug.cgi?id=1852304).
+* Datastore used in storage classes / OCP config is too long: [#1884202](https://bugzilla.redhat.com/show_bug.cgi?id=1884202).
+* Users don't enable `disk.enableUUID=TRUE`: [#1434709](https://bugzilla.redhat.com/show_bug.cgi?id=1434709) [1850984](https://bugzilla.redhat.com/show_bug.cgi?id=1850984) [1727991](https://bugzilla.redhat.com/show_bug.cgi?id=1727991).
+* Kubelet is configured without `--cloud-provider=vsphere`: [1632729](https://bugzilla.redhat.com/show_bug.cgi?id=1632729).
+* Storage Distributed Resource Scheduler (DRS) is not supported.
```

Comment:
I've been looking at some conversations around our CI env and as far as I could tell, the AWS VMC product that we are using for the CI clusters, enables DRS.

Where does this "not supported" come from and is this going to cause an issue for the CI?

Reply:
@JoelSpeed yeah DRS is enabled, not sure we could even disable it in VMC.
Since early September
Total vMotion Migrations: 7055

Reply:
**Storage** DRS is _not_ enabled.

Reply:
This highlights my lack of awareness of vSphere, I didn't realise there were two different types of DRS!
So if I understand, there is DRS for VMs and DRS for Storage? We have the VM one working and that's not causing issues? But Storage DRS isn't on in VMC?

Reply:
> This highlights my lack of awareness of vSphere, I didn't realise there were two different types of DRS!
> So if I understand, there is DRS for VMs and DRS for Storage? 

Right

> We have the VM one working and that's not causing issues? 

Don't get me wrong there are still openshift-test failures in VMC. We should add more montoring to determine if vmotion would be a potential cause of a test failure. I tend to think there are more issues with the vSphere cluster is busy.

I did create this script to determine if there was a correlation between test failures and vmotion 
https://gist.github.com/jcpowermac/9246a8030afe8af15cc2e913d88ca3fb
need to do more testing with it to have a better answer.

> But Storage DRS isn't on in VMC?

Right, I haven't found good documentation for vSAN with SDRS but from comments it doesn't seem to be supported from VMware.